src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheckTest.java

```diff
@@ -356,6 +356,46 @@ public void testCacheWhenUrlExternalResourceContentDoesNotChange() throws Except
         verify(checker, pathToEmptyFile, pathToEmptyFile, expected);
     }
 
+    @Test
+    public void testPathRegexMatches() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("file", getResourcePath("import-control_one.xml"));
+        checkConfig.addAttribute("path", "^.*[\\\\/]src[\\\\/]test[\\\\/].*$");
+        final String[] expected = {"5:1: " + getCheckMessage(MSG_DISALLOWED, "java.io.File")};
+
+        verify(checkConfig, getPath("InputImportControl.java"), expected);
+    }
+
+    @Test
+    public void testPathRegexMatchesPartially() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("file", getResourcePath("import-control_one.xml"));
+        checkConfig.addAttribute("path", "test");
+        final String[] expected = {"5:1: " + getCheckMessage(MSG_DISALLOWED, "java.io.File")};
+
+        verify(checkConfig, getPath("InputImportControl.java"), expected);
+    }
+
+    @Test
+    public void testPathRegexDoesntMatch() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("file", getResourcePath("import-control_one.xml"));
+        checkConfig.addAttribute("path", "^.*[\\\\/]src[\\\\/]main[\\\\/].*$");
+        final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
+
+        verify(checkConfig, getPath("InputImportControl.java"), expected);
+    }
+
+    @Test
+    public void testPathRegexDoesntMatchPartially() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("file", getResourcePath("import-control_one.xml"));
+        checkConfig.addAttribute("path", "main");
```

Comment:
You have to be careful with single words in a regular expression directory name match.
I did something similar with `RegExpOnFileName` and all CI worked fine, but it failed locally for @romani because of his specific directory structure naming outside of the checkstyle project. There are no slashes in this name so we could match any full word that happens to have `main` in it. Ex: **main**tain, **main**tenance
Do we need these `MatchPartially` tests for coverage? If not, and romani doesn't disagree, I think we should just remove them.